stdout-log/handler.lua

```diff
@@ -16,24 +12,56 @@ function StdoutLogHandler:new()
   StdoutLogHandler.super.new(self, "stdout-log")
 end
 
+-- Returns a copy of the input table filtered to only those keys present
+-- in allowed_keys. The corresponding values will either be:
+--     * the same as in the input
+--     * the result of calling the function value from special_cases
+--       corresponding to the same key (letting us do further filtering
+--       of nested tables)
+function filterTable(table, allowed_keys, special_cases)
+  local filteredTable = {}
+
+  for key, value in pairs(table) do
+    for _, allowed_key in pairs(allowed_keys) do
+      if key == allowed_key then
+        local special_cased = false
+
+        for special_case_key, special_case_handler in pairs(special_cases) do
+          if key == special_case_key then
+            filteredTable[key] = special_case_handler(value)
+            special_cased = true
+            break
+          end
+        end
+
+        if not special_cased then
+          filteredTable[key] = value
+        end
+        break
+      end
+    end
+  end
+
+  return filteredTable
+end
+
+function filterSerializedFields(serialized, conf)
```

Comment:
this pseudo-recursive version is my second attempt at the filtering - my first felt wayyy too verbose and tricky to maintain (https://github.com/bneutra/kong-stdout-log/pull/1/commits/fa14b92f9f526de8467d819513783df231723852#diff-079539671b7c56c77de8393806aa405a)

I know recursion can be somewhat controversial, but this should only ever hit the max stack depth of 3